src/main.rs

```diff
@@ -747,7 +747,6 @@ fn backtrace(
         if top_exception.is_none() {
```

Comment:
I'm not sure the code above this line applies anymore. the code inside the if checks for overflow but doesn't do special logic around unwinding the frame. The frame is unwound normally using the logic in line 797.

probably the code around this can be simplified now that we put a breakpoint on HardFaultTrampoline

Reply:
Yeah, seems like a good idea. I do find the code in probe-run to be a bit hard to follow sometimes.